### PR TITLE
Add predicate expressions

### LINKED ISSUE #28: Condition check if variable is defined or not

### PR DESCRIPTION
**Pull request by davidhorak**
## Changes
- Add predicate expression type.
- Add UNDEFINED predicate expression.
 ```["UNDEFINED", "$RefA"]``` evaluated as true, if the RefA does not exist in the context or if set to **undefined**.

### Test Coverage
- [x] unit

#### References
Closes #28

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -350,13 +359,26 @@ engine.evaluate(['SUFFIX', "establishment", "ment"]); // true
 engine.evaluate(['SUFFIX', "establish", "ment"]); // false
 ```
 
+### Predicate Expressions
+
+#### Undefined
+Expression format: ```["UNDEFINED", ```[Reference Operand](#reference)```]```.
+```json
+["UNDEFINED", "$RefA"]
```
Comment (rthreei) [inline readme.md:367]: Could this be achieved with:
```json
["==", "$RefA", "UNDEFINED"]
```
?

Comment (rthreei) [reply readme.md:367]: 👍

**Comment by lilyvanoekel** [comment]
So many changed files, was that a line break character conversion?

**Comment by davidhorak** [comment]
> So many changed files, was that a line break character conversion?

Couple small comments updates, but mainly the introduction of "predicate" expression. rest of the files are updated d.ts files, TS type files published within the JS library.

**Pull request merged**